:octopus: Changes required for CI scripts

neilbmclaughlin opened the pull request:
The CI scripts need to build an answers file for use with the `rancher
up` command and only have the repo names available.

The paramater names have been changed to match the repo names.

neilbmclaughlin committed:
:octopus: Changes required for CI scripts

The CI scripts need to build an answers file for use with the `rancher
up` command and only have the repo names available.

The paramater names have been changed to match the repo names.
The - in repo names is invalid for parameters so substituted for _

beseven reviewed (approved):
LOTM

st3v3nhunt reviewed (changes requested):
These changes have been made in the `0` version. Should they have been done in `1`?
This branch is behind `master` and is probably worth rebasing.

neilbmclaughlin commented:
Not sure of the versioning best practice. I guess we should increment the version each time, so perhaps, these changes should be in '2'.

Also, not sure if we can or should be doing this through GitHub's release tagging in order to avoid hard coding versions in the URL's used in the CI scripts.

Issue raised [here](https://github.com/nhsuk/ci-deployment/issues/8)

neilbmclaughlin commented:
As discussed, I'm going to move the mongo db changes in version 2 (folder 1) into version 1 (folder 0) so we can carry on using the hard coded /0 in the curl path in the CI scripts.

This means we can not easily create a stack with the old services but it is not likely we will need to do this. Should we need to we can either roll back or create a branch for the pre-mongo version of the templates and add a new catalog entry using the URL for the branch.

neilbmclaughlin committed:
:hocho: Remove version '1' directory

Following discussions regarding versioning we are going to stick
with not using directory structures to implement versioning as we
do not need to support multiple versions at this point. All changes
will be made in the '0' directory and rely on git for version
management

neilbmclaughlin committed:
:package: Add variable to support pharmacy db

neilbmclaughlin committed:
:wrench: Use correct formatting in racher compose

neilbmclaughlin commented:
@st3v3nhunt - changed to remove the c2s-pharmacy-finder  '1' directory after ensuring any commits in 1 were reflected in 0. This should be to merge now.

st3v3nhunt reviewed (changes requested):
Looks good.

Question about the consistency of the description and label of the questions in the rancher-compose for the docker image tags. There is a mix of `_`, ` ` and `-` used. I know it isn't important from the system's POV but can these be consistent between the 3 questions for the images of the api, web app and db?

beseven commented:
Re-looked at changes and they seem sensible to me!

st3v3nhunt commented:
@neilbmclaughlin Is this PR still valid? If so can it be rebased please and the last comment addressed. Thanks!

neilbmclaughlin committed:
:bug: Fix variable names to match convention

neilbmclaughlin commented:
The latest version of the CI PR script now handles multiple template versions in order to get the latest.
